src/dataconnect/fileUtils.ts

```diff
@@ -152,3 +160,48 @@ export async function getPlatformFromFolder(dirPath: string) {
   // because we found indicators for multiple platforms.
   return Platform.MULTIPLE;
 }
+
+export async function resolvePackageJson(
+  packageJsonPath: string,
+): Promise<PackageJSON | undefined> {
+  let validPackageJsonPath = packageJsonPath;
+  if (!packageJsonPath.endsWith("package.json")) {
+    validPackageJsonPath = path.join(packageJsonPath, "package.json");
+  }
+  validPackageJsonPath = path.resolve(validPackageJsonPath);
+  try {
+    return JSON.parse((await fs.readFile(validPackageJsonPath)).toString());
+  } catch {
+    return undefined;
+  }
+}
+
+export const SUPPORTED_FRAMEWORKS: (keyof SupportedFrameworks)[] = ["react"];
+function lookForFramework(
+  key: "dependencies" | "devDependencies",
+  packageJson: PackageJSON,
+  frameworksUsed: (keyof SupportedFrameworks)[],
+) {
+  if (!packageJson[key]) {
+    return frameworksUsed;
+  }
+  Object.keys(packageJson[key]!).forEach((dep) => {
```

Comment:
This currently mutates the frameworksUsed arg - would prefer to not have side effect driven code. Maybe simplify to something like:

```
function getFrameworksFromPackageJson(  
  packageJson: PackageJSON,
): (keyof SupportedFrameworks)[] {
  const dependencies = Object.keys(packageJson.dependencies?? {});
  const devDependencies = Object.keys(packageJson.devDependencies ?? {})

  const matches = [...dependencies, ...devDependencies].filter(
     dep => SUPPORTED_FRAMEWORKS.includes(dep));
   // Using set to deduplicate saves us from the extra .includes check
   return [... new Set(matches)];
}
```